programs/mango-v4/tests/cases/test_force_close.rs

```diff
@@ -0,0 +1,275 @@
+use super::*;
+
+#[tokio::test]
+async fn test_force_close() -> Result<(), TransportError> {
+    let test_builder = TestContextBuilder::new();
+    let context = test_builder.start_default().await;
+    let solana = &context.solana.clone();
+
+    let admin = TestKeypair::new();
+    let owner = context.users[0].key;
+    let payer = context.users[1].key;
+    let mints = &context.mints[0..4];
+    let payer_mint_accounts = &context.users[1].token_accounts[0..4];
+
+    //
+    // SETUP: Create a group and an account to fill the vaults
+    //
+
+    let mango_setup::GroupWithTokens { group, tokens, .. } = mango_setup::GroupWithTokensConfig {
+        admin,
+        payer,
+        mints: mints.to_vec(),
+        ..GroupWithTokensConfig::default()
+    }
+    .create(solana)
+    .await;
+    let collateral_token = &tokens[0];
+    let borrow_token = &tokens[1];
+
+    // deposit some funds, to the vaults aren't empty
+    let vault_account = send_tx(
```

Comment:
use create_funded_account() for the vault, liqor and liqee; cuts down on boilerplate a good bit